src/rule_definition_tools.jl

```diff
@@ -80,16 +80,16 @@ macro scalar_rule(call, maybe_setup, partials...)
         end
     end
     if all(Meta.isexpr(partial, :tuple) for partial in partials)
-        forward_rules = Any[rule_from_partials(partial.args...) for partial in partials]
+        forward_rules = Any[rule_from_partials(promote(inputs...)[1], partial.args...) for partial in partials]
```

Comment:
Can't call `promote` as these are expressions not values

Reply:
:man_facepalming: